feat: send email from gmail account

aayushchugh opened the pull request:
### Description

created new route `POST /api/mail/:id/gmail/:email` by which we can send email to any account from users email id

### Checklist:

<!-- pull request can be still created if all tasks are not completed  -->

-   [x] This pull request follow our contribution guidelines
-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [x] I have made corresponding changes to the documentation (postman and wiki)
-   [ ] I have written tests for the code
-   [ ] I have updated .env.example file for new .env variables

### What is current behavior?

Closes #126 

### What is new behavior?

request body
```json
{
    "to": "info@multiemail.us",
    "subject": "Testing send gmail",
    "html": "<h1>This is for testing</h1>"
}
```

preview of sent email
![image](https://user-images.githubusercontent.com/69336518/198814892-33ab92e7-6ede-4602-95b9-07c03970869a.png)

aayushchugh committed:
feat: send email from gmail account

tharun634 commented on src/controllers/mail.controller.ts:
```diff
@@ -64,3 +65,58 @@ export const getEmailsFromGmailHandler = async (
+	const user = res.locals.user as User;
+
+	try {
+		const connectedAccount = user.connected_services.find((service) => service.email === email);
```
Considering we'll be extracting `connectedAccount` in many `mail.controllers` from the request, wouldn't it make sense to just make it as a middleware to prevent copy-pasting the same code?

aayushchugh replied:
I will create a new middleware than

aayushchugh replied:
done

aayushchugh committed:
refactor: new middleware `getCurrentConnectedService`

this middleware will get the current service from the email present in `req.params`

is-it-ayush commented on src/controllers/mail.controller.ts:
```diff
@@ -56,9 +47,55 @@ export const getEmailsFromGmailHandler = async (
+	req: Request<PostSendGmailSchema["params"], {}, PostSendGmailSchema["body"]>,
+	res: Response,
+) => {
+	const { to, subject, html } = req.body;
```
Hey Ayush, Great Code. However we also need to take care of sanitization. We need to sanitize the html recieved on the backend just as a safety precaution before sending it to the client. You can use the following libraries, [DOMpurify](https://www.npmjs.com/package/dompurify) & [JSdom](https://www.npmjs.com/package/jsdom). You can either use those two libraries, or take a look at this. [Isomorphic-DOMpurify](https://github.com/kkomelin/isomorphic-dompurify) which is esentially both the previous libraries combined into one. They both do the same thing. Good Luck!

aayushchugh replied:
Ok, will do this

aayushchugh replied:
@is-it-ayush added sanitization please check now

aayushchugh committed:
Merge branch 'main' into aayushchugh/issue126

aayushchugh committed:
feat: sanitize html before sending email in gmail

is-it-ayush reviewed (approved):
Good Job Ayush! Approved!
